grug/malachite/src/actors/host/host.rs

```diff
@@ -0,0 +1,560 @@
+        match message {
+            HostMsg::ConsensusReady(actor_ref) => self.consensus_ready(state, actor_ref),
+            HostMsg::GetHistoryMinHeight { reply_to } => {
+                self.get_history_min_height(state, reply_to)
+            },
+            HostMsg::StartedRound {
+                height,
+                round,
+                proposer,
+                role,
+            } => self.started_round(state, height, round, proposer, role),
+            HostMsg::GetValue {
+                height,
+                round,
+                timeout,
+                reply_to,
+            } => {
+                self.get_value(state, height, round, timeout, reply_to)
+                    .await
+            },
+            HostMsg::RestreamValue {
+                height,
+                round,
+                valid_round,
+                address,
+                value_id,
+            } => {
+                self.restream_value(state, height, round, valid_round, address, value_id)
+                    .await
+            },
+            HostMsg::ReceivedProposalPart { part, reply_to, .. } => {
+                self.received_proposal_part(state, part, reply_to).await
+            },
+            HostMsg::GetValidatorSet { reply_to, .. } => {
+                reply_to.send(Some(self.validator_set.clone()))?;
+                Ok(())
+            },
+            HostMsg::Decided {
+                certificate,
+                consensus,
+                ..
+            } => self.decided(state, certificate, consensus).await,
+            HostMsg::GetDecidedValue { height, reply_to } => {
+                self.get_decided_value(state, height, reply_to).await
+            },
+            HostMsg::ProcessSyncedValue {
+                height,
+                round,
+                validator_address,
+                value_bytes,
+                reply_to,
+            } => {
+                self.process_synced_value(
+                    state,
+                    height,
+                    round,
+                    validator_address,
+                    value_bytes,
+                    reply_to,
+                )?;
+
+                Ok(())
+            },
+            HostMsg::ExtendVote { reply_to, .. } => {
+                reply_to.send(None)?;
+                Ok(())
+            },
+            HostMsg::VerifyVoteExtension { reply_to, .. } => {
+                reply_to.send(Ok(()))?;
+                Ok(())
+            },
+        }
+    }
+
+    fn consensus_ready(
+        &self,
+        state: &mut State,
+        consensus: ConsensusRef<Context>,
+    ) -> ActorResult<()> {
+        let height = DECIDED_BLOCK
+            .keys(state, None, None, grug::Order::Descending)
+            .next()
+            .transpose()?
+            .unwrap_or_default();
+
+        state.set_consensus(consensus.clone());
+
+        // TODO: is a sleep necessary here? It's present in the example implementation
```

Comment:
I put it in the example as it was also in our other test apps. I think it was because of some timing issues (might be from testing framework), but I asked to get clarifications.